core/src/nucleus/state.rs

```diff
@@ -17,6 +18,7 @@ impl Default for NucleusStatus {
     }
 }
 
+pub type ValidationResult = Result<(), String>;
```

Comment:
More reason to rename action. This type shares the same name...

Reply:
yeah this does bug me a bit about rust, that aliases are used commonly without namespaces, so your local unique names are sort-of global unique names

Reply:
Would renaming it `HcValidationResult` solve this issue?